SU2_CFD/src/output/filewriter/CCGNSFileWriter.cpp

```diff
@@ -0,0 +1,306 @@
+ * \version 7.0.8 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../include/output/filewriter/CCGNSFileWriter.hpp"
+#include "../../../../externals/cgns/cgnslib.h"
+#include "../../../../Common/include/geometry/CGeometry.hpp"
+#include "../../../../Common/include/CConfig.hpp"
+#include "../../../include/solvers/CSolver.hpp"
+
+#include <algorithm>
+
+const string CCGNSFileWriter::fileExt = ".cgns";
+
+CCGNSFileWriter::CCGNSFileWriter(string valFileName, CParallelDataSorter *valDataSorter) :
+  CFileWriter(std::move(valFileName), valDataSorter, fileExt){}
+
+CCGNSFileWriter::~CCGNSFileWriter(){}
+
+void CCGNSFileWriter::Write_Data(){
+
+  if (!dataSorter->GetConnectivitySorted()){
+    SU2_MPI::Error("Connectivity must be sorted.", CURRENT_FUNCTION);
+  }
+
+  #ifdef HAVE_MPI
+      SU2_MPI::Barrier(MPI_COMM_WORLD);
+  #endif
+
+  startTime = SU2_MPI::Wtime();
+  int iProcessor, nProcessor = size, index;
+  unsigned long Buffer_Send_nVertex[1], *Buffer_Recv_nVertex = nullptr;
+
+  unsigned long nLocalVertex = 0, MaxLocalVertex = 0;
+  nLocalVertex   = dataSorter->GetnPoints();
+  unsigned long nPoint_Global = dataSorter->GetnPointsGlobal();
+  Buffer_Send_nVertex[0] = nLocalVertex;
+  if (rank == MASTER_NODE) Buffer_Recv_nVertex = new unsigned long[nProcessor];
+
+  SU2_MPI::Allreduce(&nLocalVertex, &MaxLocalVertex, 1,
+                       MPI_UNSIGNED_LONG, MPI_MAX, MPI_COMM_WORLD);
+  SU2_MPI::Gather(&Buffer_Send_nVertex, 1, MPI_UNSIGNED_LONG,
+                    Buffer_Recv_nVertex,  1, MPI_UNSIGNED_LONG,
+                    MASTER_NODE, MPI_COMM_WORLD);
+
+  const vector<string> fieldNames = dataSorter->GetFieldNames();
+  unsigned short nVar = fieldNames.size();
+
+  su2double *buffLocData = new su2double[MaxLocalVertex * nVar]();
+  su2double *buffGlobData = nullptr;
+
+  index = 0;
+  for (int iPoint = 0; iPoint < nLocalVertex; iPoint++) {
+      for(int varCount = 0; varCount < nVar; varCount++){
+          buffLocData[index] = dataSorter->GetData(varCount, iPoint);
+          index++;
+      }
+  }
+
+  #ifdef HAVE_MPI
+      SU2_MPI::Barrier(MPI_COMM_WORLD);
+  #endif
```

Comment:
Whenever you use SU2_MPI:: the "ifdef endif" should not be necessary.
The "ifdef" is from the days when the MPI routines were called directly rather than via the SU2_MPI wrapper.

Reply:
Thank you, I removed the `#ifded #endif`